content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,272 @@
+
+  - **transient failures** such as the Horizon instance you're relying on going down, your router resetting, AWS hiccupping, etc. which may occur rarely but are nonetheless crucial to handle.
+
+There isn't a universal way to handle HTTP errors: some are the result of user error, some are dynamically recoverable, and some are unavoidable altogether. Transient errors are even trickier to handle due to their rarity and uncertainty, but can be made easier under the assumption that the failing backends will be up to speed again Soon:tm:. 
+
+Let's dive into each of these separately: we'll go over examples, some common causes, and recommended resolution strategies. Note that different errors are manifested differently by each SDK, so we'll just use `curl` in examples to maintain uniformity.
+
+
+# Transient Errors
+The simplest transient "outage" scenario would be your machine disconnecting from the Internet:
+
+    curl: (6) Could not resolve host: horizon.stellar.org
+
+but there are innumerable other scenarios (DNS down, firewall misconfigured, AWS outage, etc. just to name a few) that can cause them.
+
+The most surefire way to recover from a random outage is to simple: *keep trying*. Regardless of which part of the pipeline goes down, properly retrying guarantees *eventual* success (just be sure to implement back-off). This will work for most query failures, but handling an outage that occurred during a transaction submission requires a little more finesse: there's uncertainty about whether or not the transaction actually made it into a ledger. Since this situation is a little more precarious, we discuss this in more detail in a [later section](#outages).
+
+# Query Errors
+There are many, *many* possible mistakes you can make while querying the Horizon API. Naturally, using an SDK will alleviate the silliest ones, but there are still plenty of ways to go wrong. To keep the example simple here, we'll just query for details about a transaction that doesn't exist (the hash is one for an impossible null transaction):
+
+<CodeExample>
+
+```bash
+curl -s --get https://horizon.stellar.org/transaction/e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 | jq
+```
+
+</CodeExample>
+
+<CodeExample>
+
+```json
+{
+  "type": "https://stellar.org/horizon-errors/not_found",
+  "title": "Resource Missing",
+  "status": 404,
+  "detail": "The resource at the url requested was not found.  This usually occurs for one of two reasons:  The url requested is not valid, or no data in our database could be found with the parameters provided."
+}
+```
+
+</CodeExample>
+
+It's worth noting that SDKs commonly classify 404s uniquely relative to other error codes (e.g. `NotFoundError` vs the generic `NetworkError` in the JavaScript SDK).
+
+## Resolutions
+TODO: Either fix the parameters (in case of user error, insert link to API reference) or try again (in case of transient error). 
+
+Are there other cases?
+
+
+
+# Transaction Submission Errors
```

Comment:
I think it's worth adding a general rule of thumb here: any response other than `200 OK` or `400 Bad Request` means that the final status of the transaction is unknown.

Also, would be great to create a section with a generic error handling algorithm in pseudocode.

Reply:
> Also, would be great to create a section with a generic error handling algorithm in pseudocode.

TBD in a future PR!